pkg/apis/monitoring/v1/alertmanager_types.go

```diff
@@ -247,6 +247,11 @@ type AlertmanagerSpec struct {
 	// If the service account has `automountServiceAccountToken: true`, set the field to `false` to opt out of automounting API credentials.
 	// +optional
 	AutomountServiceAccountToken *bool `json:"automountServiceAccountToken,omitempty"`
+	// Enable access to Alertmanager feature flags. By default, no features are enabled.
+	// Enabling features which are disabled by default is entirely outside the
+	// scope of what the maintainers will support and by doing so, you accept
+	// that this behaviour may break at any time without notice.
+	EnableFeatures []string `json:"enableFeatures,omitempty"`
```

Comment:
We should follow the same structure than for Prometheus/PrometheusAgent CRD (e.g. name + value fields) and reuse the same functions to generate the arguments list.

Reply:
Hi, thanks for the quick response, not sure I understand what should be changed, could you elaborate a bit more?
I can see that in the prometheus arguments they use a `[]monitoringv1.Argument` but in the Alertmanager crd they use a simple `[]string` for arg generation so I simply appended to the args instead.
let me know what you think.